### PR TITLE
[Return-types Fix] Update metric tensor transforms

### PR DESCRIPTION
**Pull request by eddddddy**
**Context:**
Part of the ongoing effort to update the `batch_transforms` for the new return types system.

**Description of the Change:**
Update the `metric_tensor`, `adjoint_metric_tensor`, and `classical_fisher` transforms.

### CODE REVIEW

File: pennylane/qinfo/transforms.py [hunk 1]
```diff
@@ -462,7 +466,7 @@ def circ(params):
             [2.16840434e-18, 2.81967252e-01]]]))
 
     """
-    new_qnode = _make_probs(qnode, post_processing_fn=lambda x: qml.math.squeeze(qml.math.stack(x)))
+    new_qnode = _make_probs(qnode)
```
Comment (rmoyard) [inline pennylane/qinfo/transforms.py:469]: Oh so it was also not necessary in the old return system case?

File: pennylane/transforms/metric_tensor.py [hunk 2]
```diff
@@ -351,7 +353,7 @@ def wrapper(*args, **kwargs):
                     "much more efficient to request the block-diagonal approximation directly!"
                 )
             tkwargs["approx"] = "block-diag"
-            return self.__call__(qnode, *targs, **tkwargs)(*args, **kwargs)
+            return self(qnode, *targs, **tkwargs)(*args, **kwargs)
```
Comment (rmoyard) [inline pennylane/transforms/metric_tensor.py:356]: Good catch!

File: pennylane/transforms/metric_tensor.py [hunk 3]
```diff
@@ -611,6 +613,10 @@ def processing_fn(results):
         # Get full block-diagonal tensor
         diag_mt = diag_proc_fn(diag_res)
 
+        if qml.active_return():
+            # the off diag tapes only have a single expval measurement
+            off_diag_res = [qml.math.expand_dims(res, 0) for res in off_diag_res]
```
Comment (rmoyard) [inline pennylane/transforms/metric_tensor.py:618]: Here we expect (1,) shape instead of the new () shape?

**Review by rmoyard — COMMENTED** [review]
It looks good @eddddddy 🔥 Can you confirm me that the tests were not modified? Also some questions about the changes. Happy to approve after that

**Comment by eddddddy** [comment]
@rmoyard Yes sorry, I forgot to mention that none of the test files were modified.

File: pennylane/qinfo/transforms.py (hunk 1 above)
Comment (eddddddy) [reply pennylane/qinfo/transforms.py:469]: If `post_processing_fn` is not passed, it defaults to `lambda x: qml.math.squeeze(qml.math.stack(x))`.

File: pennylane/transforms/metric_tensor.py (hunk 3 above)
Comment (eddddddy) [reply pennylane/transforms/metric_tensor.py:618]: In the new system the raw results is a list of arrays with shape `()`. In the old system it is a list of arrays with shape `(1,)`. Therefore we expand the dims in the new system to make it consistent with the old system.

File: pennylane/qinfo/transforms.py (hunk 1 above)
Comment (rmoyard) [reply pennylane/qinfo/transforms.py:469]: Ok thanks!

File: pennylane/transforms/metric_tensor.py (hunk 3 above)
Comment (rmoyard) [reply pennylane/transforms/metric_tensor.py:618]: Thanks!

**Review by rmoyard — APPROVED** [review]
Great! Approved 💯

**Comment by eddddddy** [comment]
Not a lot of changes, so I think 1 reviewer is enough.

**Pull request merged**